weblogic-azure-aks/src/main/bicep/mainTemplate.bicep

```diff
@@ -311,6 +313,26 @@ param wlsUserName string = 'weblogic'
 var _enableCustomSSL = enableCustomSSL
 var _enableAppGWIngress = enableAppGWIngress
+// We can streamline the following code with a user-defined function, but it is not supported in Partner Center.
+// For status, see https://dev.azure.com/edburns-msft/Open%20Standard%20Enterprise%20Java%20(Java%20EE)%20on%20Azure/_workitems/edit/6219
+var _objTagsByResource = {
+  '${identifier.accounts}': contains(tagsByResource, '${identifier.accounts}') ? tagsByResource['${identifier.accounts}'] : json('{}')
+  '${identifier.managedClusters}': contains(tagsByResource, '${identifier.managedClusters}') ? tagsByResource['${identifier.managedClusters}'] : json('{}')
+  '${identifier.applicationGateways}': contains(tagsByResource, '${identifier.applicationGateways}') ? tagsByResource['${identifier.applicationGateways}'] : json('{}')
+  '${identifier.registries}': contains(tagsByResource, '${identifier.registries}') ? tagsByResource['${identifier.registries}'] : json('{}')
+  '${identifier.virtualMachines}': contains(tagsByResource, '${identifier.virtualMachines}') ? tagsByResource['${identifier.virtualMachines}'] : json('{}')
+  '${identifier.virtualMachinesExtensions}': contains(tagsByResource, '${identifier.virtualMachinesExtensions}') ? tagsByResource['${identifier.virtualMachinesExtensions}'] : json('{}')
+  '${identifier.virtualNetworks}': contains(tagsByResource, '${identifier.virtualNetworks}') ? tagsByResource['${identifier.virtualNetworks}'] : json('{}')
+  '${identifier.networkInterfaces}': contains(tagsByResource, '${identifier.networkInterfaces}') ? tagsByResource['${identifier.networkInterfaces}'] : json('{}')
+  '${identifier.networkSecurityGroups}': contains(tagsByResource, '${identifier.networkSecurityGroups}') ? tagsByResource['${identifier.networkSecurityGroups}'] : json('{}')
+  '${identifier.publicIPAddresses}': contains(tagsByResource, '${identifier.publicIPAddresses}') ? tagsByResource['${identifier.publicIPAddresses}'] : json('{}')
+  '${identifier.storageAccounts}': contains(tagsByResource, '${identifier.storageAccounts}') ? tagsByResource['${identifier.storageAccounts}'] : json('{}')
+  '${identifier.vaults}': contains(tagsByResource, '${identifier.vaults}') ? tagsByResource['${identifier.vaults}'] : json('{}')
+  '${identifier.userAssignedIdentities}': contains(tagsByResource, '${identifier.userAssignedIdentities}') ? tagsByResource['${identifier.userAssignedIdentities}'] : json('{}')
+  '${identifier.dnszones}': contains(tagsByResource, '${identifier.dnszones}') ? tagsByResource['${identifier.dnszones}'] : json('{}')
+  '${identifier.workspaces}': contains(tagsByResource, '${identifier.workspaces}') ? tagsByResource['${identifier.workspaces}'] : json('{}')
+  '${identifier.deploymentScripts}': contains(tagsByResource, '${identifier.deploymentScripts}') ? tagsByResource['${identifier.deploymentScripts}'] : json('{}')
+}
```

Comment:
When reviewing this segment, I had the desire to assert that each of the `${identifier.foo}` values in this region shows up in at least one bicep file within the set of **Files changed** for this PR. I tried to use GitHub Copilot within this PR, but it failed miserably. Instead, I used Bing copilot to write this bash function that performed the assertion

```bash
review_all_refs_appear_atleast_once() {
    references_file="$1"
    files_file="$2"
    missing_variables=()

    while IFS= read -r variable; do
        found=false
        while IFS= read -r file; do
            if grep -q "$variable" "$file"; then
                found=true
                break
            fi
        done < "$files_file"
        if [ "$found" = false ]; then
            missing_variables+=("$variable")
        fi
    done < "$references_file"
    
    if [ ${#missing_variables[@]} -eq 0 ]; then
        echo "All variables are present in at least one file."
    else
        echo "The following variables are missing:"
        for variable in "${missing_variables[@]}"; do
            echo "$variable"
        done
    fi
}
```

Using this function, I found that 

```
The following variables are missing:
identifier.virtualMachinesExtensions
identifier.networkInterfaces
```

And, indeed, this appears to be true.

Why? If we don't use those identifiers, then we should not be including them in this regoin.

Reply:
Hello @edburns, thanks for your careful review! I really appreciate your approach of using AI to identify the excluded identifiers.

In the AKS offer, we use AZ CLI to create the Ubuntu VM, which is removed after building the container image.

![image](https://github.com/user-attachments/assets/1fcdfe04-ef9e-44de-a836-37e8faa89f6b)

The network interface (NIC) is created by the same command, and it automatically applies the VM's tags.

Now, we have three options:

- **Keep the current implementation**: Do not input tags for the NIC, and let it inherit the VM's tags.
- **Allow customers to set tags for the NIC**: Apply tags to both the VM and the NIC, and they share the same tags.
- **Use an ARM template to create the VM and its extension**: This would allow us to apply corresponding tags to both the VM and NIC (though this introduces a significant change, and I would suggest a separate PR for this).

I recommend keeping the current implementation, as our primary goal is to add tags to the VM, and the AZ CLI command automatically applies those tags to the NIC as well.

Regarding the VM extension, as noted, the `az vm extension set` command currently does not support adding tags.